File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public static List<String> result = new ArrayList<>();
    public static List<Character> vowels = List.of('A', 'E', 'I', 'O', 'U');
    
    public static void check(String currentWord) {
        if (currentWord.length() <= 5) {
            if (!currentWord.isEmpty()) {
                result.add(currentWord);
            }
            for (Character vowel : vowels) {
                check(currentWord + vowel);
            }
        }
    }

    public static int solution(String word) {
        check("");
        return result.indexOf(word) + 1;
    }
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.kt
class Solution {
    fun solution(word: String): Int {
        val vowels = listOf('A', 'E', 'I', 'O', 'U')
        val result = mutableListOf<String>()

        fun backtrack(currentWord: String) {
            if (currentWord.length <= 5) {
                if (currentWord.isNotEmpty()) {
                    result.add(currentWord)
                }
                for (vowel in vowels) {
                    backtrack(currentWord + vowel)
                }
            }
        }
        backtrack("")

        return result.indexOf(word) + 1
    }
}